docs/components/FlatUASTViewer.md

```diff
@@ -0,0 +1,24 @@
+Renders transformed UAST.
+
+```js
+const someUast = {
+    1: {
+        "InternalType": "CompilationUnit",
+        "StartPosition": { "Offset": 0, "Line": 1, "Col": 1 },
+        "EndPosition": { "Offset": 446, "Line": 0, "Col": 0 },
+        "Roles": ["File"],
+        "Children": [2],
+        expanded: true
+    },
+    2: {
+        "InternalType": "LineComment",
+        "Properties": { "internalRole": "comments", "text": " hello.java" },
+        "StartPosition": { "Offset": 0, "Line": 1, "Col": 1 },
+        "EndPosition": { "Offset": 13, "Line": 1, "Col": 14 },
+        "Roles": ["Comment"],
+        "Children": []
+    }
+};
+
+<FlatUASTViewer flatUast={someUast} />
```

Comment:
I disagree. This example shows how to use `FlatUASTViewer` component. It doesn't depend on raw UAST at all.
It's important to understand flat-uast if you need **any** customization (look at controlled mode example)

Reply:
Up to you I guess, but I find this example is confusing. Why would you want to use `FlatUASTViewer` here instead of `UASTViewer`? The only reason I see is because `UASTViewer` cannot be controlled, but this example is not controlled.

Reply:
Yes, you are right, this is documenting `FlatUASTViewer`. So it should also serve as an example of a use case for it. The current example shows that you can render flat UAST without mentioning that you have a transformer method to deal with it. And transforming your raw UAST to flat and rendering it with `FlatUASTViewer` offers nothing over the more simple `UASTViewer`.

You can use it like that, but you probably don't want to unless there is some other justification.

Anyway, I don't feel that strongly about it, the example is not "incorrect" code, I just though it misses some context as to when you would want to use `FlatUASTViewer` over `UASTViewer`. Let's merge as it is if you prefer it.

Reply:
Ah. So your point is not about using transformer but about describing the difference between UASTViewer and FlatUASTViewer. It's a good point!